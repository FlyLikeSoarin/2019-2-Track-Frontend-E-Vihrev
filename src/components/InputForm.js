import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const InputField = styled.input`
  border: 0;
  outline: none;
  width: calc(100% - 2px);
  font-size: 20px;
  :host {
    display: inline-block;
    border: 1px solid rgba(25, 25, 25, 0.32);
  }
`;

class InputForm extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			inputValue: '',
		};

		this.refInputField = React.createRef();
	}

	componentDidMount() {
		if (this.props.isInFocus) {
			this.focus();
		}
	}

	componentDidUpdate() {
		if (this.props.isInFocus) {
			this.focus();
		}
	}

	onSubmit(evt) {
		evt.preventDefault();
		this.props.submitHandler(this.state.inputValue);
		this.reset();
	}

	onChange(evt) {
		this.setState({inputValue: evt.target.value});
	}

	reset() {
		this.setState({inputValue: ''});
	}

	focus() {
		this.refInputField.current.focus();
	}

	render() {
		return (
			<form onSubmit={(evt) => this.onSubmit(evt)}>
				<InputField
					style={this.props.style}
					name={this.props.name}
					placeholder={this.props.placeholder}
					value={this.state.inputValue}
					onChange={(evt) => this.onChange(evt)}
					ref={this.refInputField}
					autoFocus={this.props.isInFocus ? 'autofocus' : false}
					onBlur={this.props.onFocusOut}
				/>
			</form>
		);
	}
}

InputForm.defaultProps = {
	style: {},
	placeholder: '',
	isInFocus: false,
	onFocusOut: ()=>{},
};

InputForm.propTypes = {
	style: PropTypes.objectOf(PropTypes.string),
	name: PropTypes.string.isRequired,
	placeholder: PropTypes.string,
	isInFocus: PropTypes.bool,
	onFocusOut: PropTypes.func,
	submitHandler: PropTypes.func.isRequired,
};



export default InputForm;
