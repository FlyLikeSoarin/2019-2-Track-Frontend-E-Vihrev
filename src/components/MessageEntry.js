import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';
// import logo from '../assets/logo.svg';
// import backImg from '../assets/back.png';
// import menuImg from '../assets/menu.png';
// import searchImg from '../assets/search-icon.png';
// import noUserIcon from '../assets/no-user-icon.png';

const MessageOuterBox = styled.div`
	display: flex;
	flex-direction: column;
	font-size: 20px;
`;

const fadein = keyframes`
  from { opacity: 0; }
  to   { opacity: 1; }
`;

const WidthHolder = styled.div`
	max-width: 80%;
	overflow-wrap: break-word;
	/* background-color: blue; */
	/* display: flex; */
	/* flex-direction: column; */
	align-self: flex-end;
	animation: fadein 0.6s;
`;

const MessageBox = styled.div`
	position: relative;
	right: 0px;
	/* flex-shrink: 1; */
	/* align-self: flex-end; */
	background-color: #f3e5f5;
	margin: 0.15em;
	border: 0.05em;
	padding: 0.15em;
	padding-left: 0.3em;
	padding-right: 0.3em;
	border-style: solid;
	border-color: ;
	display: flex;
	flex-direction: column;
	animation: ${fadein} 0.5s;
`;

const MessageText = styled.div`
	word-wrap: word-brake;
	margin: 0.1em;
`;

const MessageDate = styled.div`
	text-align: right;
	color: grey;
	font-size: 0.75em;
`;

const MessageName = styled.div`
	align-self: flex-start;
	display: none;
	color: grey;
	font-size: 0.75em;
`;

function MessageEntry({ name, text, timestamp }) {
	return (
		<MessageOuterBox>
			<WidthHolder>
				<MessageOuterBox>
					<MessageBox>
						<MessageName> {name} </MessageName>
						<MessageText> {text} </MessageText>
						<MessageDate> {timestamp} </MessageDate>
					</MessageBox>
				</MessageOuterBox>
			</WidthHolder>
		</MessageOuterBox>
	);
}

MessageEntry.propTypes = {
	name: PropTypes.string.isRequired,
	text: PropTypes.string.isRequired,
	timestamp: PropTypes.string.isRequired,
};

export default MessageEntry;
