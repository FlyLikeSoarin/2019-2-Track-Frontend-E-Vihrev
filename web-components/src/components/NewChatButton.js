const newChatImg = require('../icons/new-chat.png');

const template = document.createElement('template');
template.innerHTML = `
<style>
  .new-chat {
    font-size: 20px;
    display: flex;
    flex-direction: row;
    margin: 10px;
  }

  .input {
    display: none;
    align-self: start;
    margin: 0.5em;
    border: 0.2em;
    border-left: 0.5em;
    border-right: 0.5em;
    border-radius: 0.5em;
    border-style: solid;
    border-color: black;
  }

  img {
    height: 70px;
    weight: 70px;
  }
</style>

<div class="new-chat">
    <img alt="New chat">
    <div class="input">
      <form class="recipient-form">
        <form-input name="message-text" placeholder="Recipient..."></form-input>
      </form>
    </div>
</div>
`;

class NewChatButton extends HTMLElement {
  constructor() {
    super();
    this.shadowRoot = this.attachShadow({ mode: 'open' });
    this.shadowRoot.appendChild(template.content.cloneNode(true));


    this.$img = this.shadowRoot.querySelector('img');
    this.$img.src = newChatImg;
    this.$img.addEventListener('click', this.onClick.bind(this));

    this.$input = this.shadowRoot.querySelector('.input');
    this.$form = this.shadowRoot.querySelector('form-input');

    this.$input.style.display = 'none';

    this.$form.addEventListener('submit', this.onSubmit.bind(this));
    this.$form.addEventListener('keypress', this.onKeyPress.bind(this));

    this.newChatCallback = null;
  }

  setNewChatCallback(callback) {
    this.newChatCallback = callback;
  }

  onClick() {
    if (this.$input.style.display === 'none') {
      this.$form.reset();
      this.$form.focusOn();
      this.$input.style.display = 'initial';
    } else {
      this.$input.style.display = 'none';
    }
  }

  onSubmit() {
    this.newChatCallback(this.$form.value);

    this.$form.reset();
    this.$input.style.display = 'none';
  }

  onKeyPress(event) {
    if (event.keyCode === 13) {
      this.$form.dispatchEvent(new Event('submit'));
    }
  }
}

customElements.define('new-chat-button', NewChatButton);
