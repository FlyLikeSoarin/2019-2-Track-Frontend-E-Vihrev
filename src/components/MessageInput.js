import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import InputForm from './InputForm';

const Container = styled.div`
	display: flex;
	flex-direction: row;
	background-color: #333;
`;

const FlexGrow = styled.div`
	flex-grow: 1;
`;

const Button = styled.div`
	border-radius: 12px;
	margin: 3px 0px 3px 6px;
	padding-left: 5px;
	background-color: #8e24aa;
	:hover {
		background-color: #ae44ca;
	}
	cursor: pointer;
`;

const FileInput = styled.input`
	width: 2px;
	height: 2px;
	opacity: 0;
	position: absolute;
	z-index: 999;
`;

const FileLabel = styled.label`
	font-size: 20px;
	margin: 3px 0px 3px 6px;
	padding: 2px 5px;
	font-weight: 700;
	color: white;
	background-color: #8e24aa;
	display: inline-block;
	:hover {
		background-color: #ae44ca;
	}
	cursor: pointer;
`;

class MessageInput extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			top: -400,
			left: -400,
		};

		const { submitHandler } = props;

		this.formRef = React.createRef();
		this.submitHandlerBounded = submitHandler.bind(this);
		this.startRecordingBounded = this.startRecording.bind(this);
		this.stopRecordingBounded = this.stopRecording.bind(this);
		this.onChangeHandlerBounded = this.onChangeHandler.bind(this);
		this.sendGeolocationBounded = this.sendGeolocation.bind(this);
		this.onDragOverBounded = this.onDragOver.bind(this);
	}

	onChangeHandler(e) {
		const { files } = e.target;

		if (FileReader && files && files.length) {
			const fr = new FileReader();
			fr.onload = () => {
				this.submitHandlerBounded(`/img_${fr.result}`);
			};
			fr.readAsDataURL(files[0]);
		}

		const formData = new FormData(this.formRef.current);

		const xhr = new XMLHttpRequest();
		xhr.open('POST', 'https://tt-front.now.sh/upload', true);
		xhr.send(formData);
	}

	onDragOver(e) {
		e.preventDefault();
		e.stopPropagation();
		const x = e.pageX;
		const y = e.pageY;

		this.setState({ top: y - 1, left: x - 1 });
	}

	startRecording() {
		if (navigator.mediaDevices) {
			navigator.mediaDevices.getUserMedia({ audio: true }).then((stream) => {
				this.mediaRecorder = new MediaRecorder(stream);
				this.mediaRecorder.start();

				const audioChunks = [];
				this.mediaRecorder.addEventListener('dataavailable', (event) => {
					audioChunks.push(event.data);
				});

				this.mediaRecorder.addEventListener('stop', () => {
					const audioBlob = new Blob(audioChunks);
					const audioUrl = URL.createObjectURL(audioBlob);

					this.submitHandlerBounded(`/aud_${audioUrl}`);

					const formData = new FormData();
					formData.append('audio', audioBlob);
					const xhr = new XMLHttpRequest();

					xhr.open('POST', 'https://tt-front.now.sh/upload', true);
					xhr.send(formData);
				});
			});
		}
	}

	stopRecording() {
		if (this.mediaRecorder !== undefined) {
			this.mediaRecorder.stop();
		}
	}

	sendGeolocation() {
		if (navigator.geolocation) {
			navigator.geolocation.getCurrentPosition((position) => {
				const { latitude } = position.coords;
				const { longitude } = position.coords;
				const url = `https://www.openstreetmap.org/#map=18/${latitude}/${longitude}`;
				this.submitHandlerBounded(`/pos_${url}`);
			});
		}
	}

	render() {
		const { top, left } = this.state;

		const inputFormStyle = {
			borderStyle: 'solid',
			borderTopWidth: '2px',
			borderColor: '#333',
		};

		const fileInputStyle = {
			top: `${top.toString()}px`,
			left: `${left.toString()}px`,
		};

		return (
			<>
				<Container>
					<FlexGrow>
						<InputForm
							name="message-text"
							placeholder="Message..."
							submitHandler={this.submitHandlerBounded}
							style={inputFormStyle}
						/>
					</FlexGrow>
					<>
						<Button
							onMouseDown={this.startRecordingBounded}
							onMouseUp={this.stopRecordingBounded}
						>
							<span role="img" aria-label="Voice Message">
								&#127908;
							</span>
						</Button>
					</>
					<>
						<Button onClick={this.sendGeolocationBounded}>
							<span role="img" aria-label="Location">
								&#128204;
							</span>
						</Button>
					</>
					<form
						action="https://tt-front.now.sh/upload"
						method="post"
						ref={this.formRef}
					>
						<FileInput
							type="file"
							name="image"
							id="image"
							onChange={this.onChangeHandlerBounded}
							style={fileInputStyle}
						/>

						<FileLabel htmlFor="image" onDragOver={this.onDragOverBounded}>
							UPLOAD IMAGE
						</FileLabel>
					</form>
				</Container>
			</>
		);
	}
}

MessageInput.propTypes = {
	submitHandler: PropTypes.func.isRequired,
};

export default MessageInput;
