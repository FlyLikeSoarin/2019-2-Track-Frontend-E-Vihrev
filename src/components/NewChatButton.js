import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';
import newChatImg from '../assets/new-chat.png';
import InputForm from './InputForm';

const NewChat = styled.div`
	font-size: 20px;
	display: flex;
	flex-direction: row;
	margin: 15px;
`;

const NewChatInput = styled.div`
	transition: 1s;

	display: flex;
	align-self: start;
	margin: 0.7em;
	border: 0.2em;
	border-left: 0.3em;
	border-right: 0.3em;
	border-style: solid;
	border-color: black;
	width: 0em;

	&.selected {
		transition: 1s;
		right: 0em !important;
		width: 15em !important;
	}
`;

const pulse = keyframes`
  from {
    box-shadow: 0 0 0 0 rgba(142, 36, 170, 0.4);
  }
  70% {
    box-shadow: 0 0 0 25px rgba(142, 36, 170, 0);
  }
  to {
    box-shadow: 0 0 0 0 rgba(142, 36, 170, 0);
  }
`;

const IconHolder = styled.div`
	height: 70px;
	position: relative;
	right: -2em;
	transition: 1s;
	border-radius: 50%;
	background: #cca92c;
	cursor: pointer;
	box-shadow: 0 0 0 rgba(142, 36, 170, 0.5);

	&:hover {
		animation: ${pulse} 2s infinite;
	}

	&.selected {
		transition: 1s;
		transform: rotate(-180deg);
		right: 0em !important;
	}
`;

const IconImg = styled.img`
	position: relative;
	right: 0px;
	height: 70px;
	weight: 70px;
`;

class NewChatButton extends React.Component {
	constructor(props) {
		super(props);
		this.refIconHolder = React.createRef();
		this.refNewChatInput = React.createRef();
		this.state = {
			isInFocus: false,
		};

		this.onFocusOutBounded = this.onFocusOut.bind(this);
		this.onClickBounded = this.onClick.bind(this);
		this.submitHandlerBounded = this.submitHandler.bind(this);
	}

	onFocusOut(evt) {
		this.Deactivate();
	}

	onClick() {
		this.Activate();
	}

	Activate() {
		this.refIconHolder.current.classList.add('selected');
		this.refNewChatInput.current.classList.add('selected');
		this.setState({ isInFocus: true });
	}

	Deactivate() {
		this.refIconHolder.current.classList.remove('selected');
		this.refNewChatInput.current.classList.remove('selected');
		this.setState({ isInFocus: false });
	}

	submitHandler(value) {
		const { newChatHandler } = this.props;
		newChatHandler(value);
		this.Deactivate();
	}

	render() {
		const { isInFocus } = this.state;
		const InputFormStyle = {
			transition: '1s',
			width: isInFocus ? '15em' : '0em',
			borderColor: 'black',
		};

		return (
			<NewChat>
				<IconHolder ref={this.refIconHolder}>
					<IconImg
						alt="New chat"
						src={newChatImg}
						onClick={this.onClickBounded}
					/>
				</IconHolder>
				<NewChatInput ref={this.refNewChatInput}>
					<InputForm
						style={InputFormStyle}
						name="message-text"
						placeholder="Recipient..."
						submitHandler={this.submitHandlerBounded}
						isInFocus={isInFocus}
						onFocusOut={this.onFocusOutBounded}
					/>
				</NewChatInput>
			</NewChat>
		);
	}
}

NewChatButton.propTypes = {
	newChatHandler: PropTypes.func.isRequired,
};

export default NewChatButton;
