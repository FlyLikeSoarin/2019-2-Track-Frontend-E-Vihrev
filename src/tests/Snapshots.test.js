import React from 'react';
import renderer from 'react-test-renderer';

import InputForm from '../components/InputForm/';
import NewChatButton from '../components/NewChatButton';
import MessageEntry from '../components/MessageEntry';
import MessageInput from '../components/MessageInput';

it('InputForm renders correctly', () => {
	const tree = renderer
		.create(<InputForm name="snapshot-input" submitHandler={() => {}} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('NewChatButton renders correctly', () => {
	const tree = renderer
		.create(<NewChatButton newChatHandler={() => {}} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('MessageEntry renders correctly', () => {
	const tree = renderer
		.create(
			<MessageEntry
				isFromUser={false}
				username="user"
				text="snapshot"
				timestamp="00:00:00"
			/>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('MessageInput renders correctly', () => {
	const tree = renderer
		.create(<MessageInput submitHandler={() => {}} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
