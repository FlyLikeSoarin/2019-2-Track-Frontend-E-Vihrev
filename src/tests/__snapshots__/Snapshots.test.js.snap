// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoFocus={false}
    className="css-jrnoqf"
    name="snapshot-input"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder=""
    style={Object {}}
    type="text"
    value=""
  />
</form>
`;

exports[`MessageEntry renders correctly 1`] = `
<div
  className="css-1v2921"
>
  <div
    className="css-ecsmvg"
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
  >
    <div
      className="css-1v2921"
    >
      <div
        className="css-fi4w0v"
      >
        <div
          className="css-p6lagf"
        >
           
          user
           
        </div>
        <div
          className="css-udf8nl"
        >
           
          snapshot
           
        </div>
        <div
          className="css-1mo3uph"
        >
           
          00:00:00
           
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MessageInput renders correctly 1`] = `
<div
  className="css-11pogay"
>
  <div
    className="css-q0vwmm"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        autoFocus={false}
        className="css-jrnoqf"
        name="message-text"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Message..."
        style={
          Object {
            "borderColor": "#333",
            "borderStyle": "solid",
            "borderTopWidth": "2px",
          }
        }
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="css-kianx0"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <span
      aria-label="Voice Message"
      role="img"
    >
      ðŸŽ¤
    </span>
  </div>
  <div
    className="css-kianx0"
    onClick={[Function]}
  >
    <span
      aria-label="Location"
      role="img"
    >
      ðŸ“Œ
    </span>
  </div>
  <form
    action="https://tt-front.now.sh/upload"
    method="post"
  >
    <input
      className="css-zfhggc"
      id="image"
      name="image"
      onChange={[Function]}
      style={
        Object {
          "left": "-400px",
          "top": "-400px",
        }
      }
      type="file"
    />
    <label
      className="css-1lxjbmj"
      htmlFor="image"
      onDragOver={[Function]}
    >
      UPLOAD IMAGE
    </label>
  </form>
</div>
`;

exports[`NewChatButton renders correctly 1`] = `
<div
  className="css-kmkywd"
>
  <div
    className="css-1fkr3uu"
  >
    <img
      alt="New chat"
      className="css-qpjx77"
      onClick={[Function]}
      src="new-chat.png"
    />
  </div>
  <div
    className="css-1qnfcaf"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        autoFocus={false}
        className="css-jrnoqf"
        name="message-text"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Recipient..."
        style={
          Object {
            "borderColor": "black",
            "transition": "1s",
            "width": "0em",
          }
        }
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
